include/odp/api/spec/traffic_mngr.h

```diff
@@ -379,6 +383,24 @@ typedef struct {
 	 * only as many number of WFQ groups as number of priorities.
 	 */
 	uint8_t max_wfq_groups_per_node;
+
+	/** tm_queue_threshold indicates support for threshold profile on a
+	 * TM queue. When TRUE, users can set/clear/update threshold profile
+	 * on a TM queue. When false clear none of it is supported.
+	 */
+	odp_bool_t tm_queue_threshold;
+
+	/** tm_queue_query_flags indicates supported types of TM queue query.
+	 * Types of TM queue query are same as query_flags that are passed to
+	 * odp_tm_queue_query(), odp_tm_priority_query() and
+	 * odp_tm_total_query(). When zero, none of the queue query API's are
+	 * supported. When non-zero, only the only supported types of passed
+	 * query_flags are taken into account and corresponding fields updated.
+	 *
+	 * @see ODP_TM_QUERY_PKT_CNT, ODP_TM_QUERY_BYTE_CNT,
+	 * ODP_TM_QUERY_THRESHOLDS.
+	 */
+	uint32_t tm_queue_query_flags;
```

Comment:
The existing API has this text in the query functions: "The query_flags indicate whether or not packet counts, byte counts or both are being requested.  It is an error to request neither. The implementation may still return both sets of counts regardless of query_flags if the cost of returning all the counts is comparable to the cost of checking the query_flags."

Now that there could be implementations that do not support the counts, this text probably should be changed. e.g. "The query flags indicate which information is begin requested. The implementation may choose to return additional information that was not requested"